app/src/main/java/com/example/trialio/controllers/ExperimentManager.java

```diff
@@ -234,6 +234,21 @@ public void onFailure(@NonNull Exception e) {
      * @return Task that indicates when the delete is complete
      */
     public Task<Void> deleteExperiment(String experimentId) {
+
+        // remove experimentID from the subscribed list of all users.
+        UserManager userManager = new UserManager();
```

Comment:
I'm worried this is going to break some tests due to the coupling between UserManager and ExperimentManager. UserManager is created here with the default collection, which won't be the same as the test collection. Do the tests pass with this?

Reply:
"Do you think we should edit the tests so that they set the collection path with the method": yeah I think this would work. It's what we do in the UI tests already, right? Ideally, it would nice to think of a new design for this as it seems kinda dirty, but I get why you're doing it and it does seem easier this way. Passing in the optional UserCollection name might work too.

Reply:
Maybe we could pass the UserManager itself into the deleteExperiment method? If a UserManager is given, delete all associated subs. If UserManager null, do nothing. That would make it less dependent on internal parts of the method

Reply:
I'm leaning towards the latter option now. That way there it is very explicit which UsermManager is being used (if any), and there isn't any sneaky stuff happening under the hood.

Reply:
yeah I like that, I'll do that for the other deletes too